.ci/cico_build_and_deploy_custom_rh-che.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+set -x
+set -e
+set +o nounset
+
+/usr/sbin/setenforce 0
+
+function retrieveActiveToken() {
+  CURL_OUTPUT=$(curl -H "Content-Type: application/json" -X POST -d '{"refresh_token":"'$KEYCLOAK_TOKEN'"}' https://auth.prod-preview.openshift.io:8443/api/token/refresh)
+  ACTIVE_TOKEN=$(echo $CURL_OUTPUT | jq --raw-output ".token | .access_token")
+  if [[ -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_USERNAME}" ]]; then
+    empty_credentials="OSIO username is empty, "
+  fi
+  if [[ -z "${RH_CHE_AUTOMATION_CHE_PREVIEW_PASSWORD}" ]]; then
+    empty_credentials=${empty_credentials}"OSIO password is empty, "
+  fi
+  if [[ -z "${ACTIVE_TOKEN}" ]]; then
+    empty_credentials=${empty_credentials}"Keycloak token is empty"
+  fi
+  if [[ ! -z "${empty_credentials}" ]]; then
+    echo ${empty_credentials}
+    exit 1
+  fi
+  if [[ $(curl -X GET -H "Authorization: Bearer ${ACTIVE_TOKEN}" https://auth.prod-preview.openshift.io:8443/api/token?for= \
+     |  grep access_token | wc -l) -ne 1 ]]; then
+    echo "Keycloak token is expired"
+    exit 1
+  else
+    echo "Keycloak token is active."
+  fi
+}
+
+function tagAndPushDocker() {
+  export RH_CHE_AUTOMATION_BUILD_TAG=rh-che-automated-build-$(git rev-parse --short HEAD)
+  export CHE_DOCKER_BASE_IMAGE=eclipse/che-server:nightly-centos
+  echo "che-docker-base-image:${CHE_DOCKER_BASE_IMAGE}"
+  docker pull ${CHE_DOCKER_BASE_IMAGE}
+  docker tag ${CHE_DOCKER_BASE_IMAGE} eclipse/che-server:local
+  rm -rf ../dockerfiles/che-fabric8/eclipse-che
+  mkdir ../dockerfiles/che-fabric8/eclipse-che
+  cp -r ../assembly/assembly-main/target/eclipse-che-fabric8-1.0.0-SNAPSHOT/eclipse-che-fabric8-1.0.0-SNAPSHOT/* ../dockerfiles/che-fabric8/eclipse-che/
+  bash ../dockerfiles/che-fabric8/build.sh || exit 1
+  docker rmi eclipse/che-server:local
+  docker rmi eclipse/che-server:nightly-centos
+  docker tag eclipse/che-server:nightly rhcheautomation/${RH_CHE_AUTOMATION_BUILD_TAG}
+  docker rmi eclipse/che-server:nightly
+  docker login -u ${RH_CHE_AUTOMATION_DOCKERHUB_USERNAME} -p ${RH_CHE_AUTOMATION_DOCKERHUB_PASSWORD}
+  docker push rhcheautomation/che-server:${RH_CHE_AUTOMATION_BUILD_TAG}
+}
+
+# Getting dependencies ready
+sudo yum install dnf
+sudo dnf install --skip-broken --assumeyes \
+            docker \
+            git \
+            make \
+            jq \
+            origin-clients \
+            python3
+systemctl start docker
+pip install yq
+
+cat jenkins-env | \
+    grep -E "(KEYCLOAK|BUILD_NUMBER|JOB_NAME|RH_CHE)" | \
+    sed 's/^/export /g' | \
+    sed 's/= /=/g' | \
+    > export_env_variables
+
+source export_env_variables
+
+retrieveActiveToken
```

Comment:
@ScrewTSW got a bit lost with `retrieveActiveToken` function.
why do we need OSIO user token aka `ACTIVE_TOKEN` for pushing image to dockerhub ?

Reply:
it is to check that the account is still provisioned. and the token can be used in the deploy script arguments

Reply:
> it is to check that the account is still provisioned
Not sure we need to make this check, but even if we need it why not to do it when executing functional tests ? From one hand I understand that you want to fail-fast, but not provisioned ci account is an edge case (once provisioned ci account is not going to be revoked, so do not think we should check it )

>  and the token can be used in the deploy script arguments
Sorry, I do not understand `why` do we need this token for deployment on dev cluster ? AFAIK, it is only needed for performing tests so, it would be cleaner to obtain it when we need it, and not just in case. Here is how it is done on che-starter side [1] token is obtained and set as env var only when tests are going to be executed. So, for me it sounds like `cico_build_and_deploy_custom_rh-che.sh` is not the right place for obtaining osio token, since it is not needed for build & deploy on dev cluster

[1] https://github.com/redhat-developer/che-starter/blob/master/cico_run_tests.sh#L32